pipeline/transform/users.py

```diff
@@ -30,3 +31,13 @@ def convert_users_from_df(df: pd.DataFrame) -> List[User]:
     field_names = [f.name for f in dataclasses.fields(User) if f.name in cols]
     user_dicts = df[field_names].to_dict(orient="records")
     return [User(**u) for u in user_dicts]
+
+
+def convert_users_into_hash(df: pd.DataFrame) -> Dict[UserId, User]:
+    cols = set(df.columns)
+    # Parse raw community membership records into dataclasses
```

Comment:
This works! No need to change anything.

In the future, we would just refactor our extract tasks to return the dataclass objects, that way we don't have to bother with converting dataframes.